internal/cli/import_yaml.go

```diff
@@ -0,0 +1,221 @@
+		TokenLifetime                             int    `yaml:"token_lifetime,omitempty"`
+		TokenLifetimeForWeb                       int    `yaml:"token_lifetime_for_web,omitempty"`
+		SigningSecret                             string `yaml:"signing_secret,omitempty"`
+	} `yaml:"resourceServers,omitempty"`
+	Roles []struct {
+		Name        string `yaml:"name"`
+		Description string `yaml:"description"`
+		Permissions []struct {
+			PermissionName           string `yaml:"permission_name"`
+			ResourceServerIdentifier string `yaml:"resource_server_identifier"`
+		} `yaml:"permissions,omitempty"`
+	} `yaml:"roles,omitempty"`
+}
+
+func ParseYAML(yamlPath string, config *ImportConfig) (*TenantConfig, error) {
+	yamlData, err := ioutil.ReadFile(yamlPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading yaml file: %v ", err)
+	}
+
+	t := &TenantConfig{}
+
+	err = yaml.Unmarshal(yamlData, t)
+	if err != nil {
+		return nil, fmt.Errorf("error Unmarshaling yaml: %v ", err)
+	}
+
+	if err = t.CheckForDuplicateNames(); err != nil {
+		return nil, err
+	}
+
+	for key, replacement := range config.Auth0KeywordReplaceMappings {
+		t.replaceClientConfig(key, replacement)
+		t.replaceResourceServersConfig(key, replacement)
+		t.replaceRolesConfig(key, replacement)
+	}
+
+	return t, nil
+}
+
+func (t *TenantConfig) replaceClientConfig(key string, replacement interface{}) {
+	for i, client := range t.Clients {
+
+		if strings.ContainsAny(client.Name, key) {
+			str := strings.ReplaceAll(client.Name, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.Clients[i].Name = str
+		}
+
+		if strings.ContainsAny(client.Description, key) {
+			str := strings.ReplaceAll(client.Description, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.Clients[i].Description = str
+		}
+
+		if strings.ContainsAny(client.TokenEndpointAuthMethod, key) {
+			str := strings.ReplaceAll(client.TokenEndpointAuthMethod, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.Clients[i].TokenEndpointAuthMethod = str
+		}
+
+		for j, item := range client.AllowedLogoutURLs {
+			if strings.ContainsAny(item, key) {
+				str := strings.ReplaceAll(item, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Clients[i].AllowedLogoutURLs[j] = str
+			}
+		}
+
+		for j, item := range client.Callbacks {
+			if strings.ContainsAny(item, key) {
+				str := strings.ReplaceAll(item, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Clients[i].Callbacks[j] = str
+			}
+		}
+
+		for j, item := range client.WebOrigins {
+			if strings.ContainsAny(item, key) {
+				str := strings.ReplaceAll(item, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Clients[i].WebOrigins[j] = str
+			}
+		}
+
+		for j, item := range client.AllowedOrigins {
+			if strings.ContainsAny(item, key) {
+				str := strings.ReplaceAll(item, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Clients[i].AllowedOrigins[j] = str
+			}
+		}
+
+		for j, item := range client.GrantTypes {
```

Comment:
I see we added more fields for clients and apis. Do we know what fields are configurable? Or can we allow replacements for all fields?

Reply:
Thanks for pointing this out. I compared with the Management API docs, and ended adding `omitempty` for `app_type` (apps) ([commit](https://github.com/auth0/auth0-cli/pull/404/commits/01280ba29f28d6a86c0af685ee724dbbc0df9d05)), `name` (APIs) ([commit](https://github.com/auth0/auth0-cli/pull/404/commits/5387d55e3faf9d1b5faaa1a183b812a265922b09)), and `description` (roles) ([commit](https://github.com/auth0/auth0-cli/pull/404/commits/f50b7fd53224046e40fc202f5d82a8a771dea1f8)).

**Apps:** only `name` is required.
<img width="692" alt="Screen Shot 2022-03-11 at 12 31 06" src="https://user-images.githubusercontent.com/5055789/157898861-7d06d5db-5801-49e2-9742-29b48682da7d.png">

**APIs:** only `identifier` is required.
<img width="695" alt="Screen Shot 2022-03-11 at 12 31 50" src="https://user-images.githubusercontent.com/5055789/157898949-e81775d7-9e14-40a8-988a-6783ca5487a2.png">

**Roles:** only `name` is required.
<img width="689" alt="Screen Shot 2022-03-11 at 12 32 15" src="https://user-images.githubusercontent.com/5055789/157899544-d8e1c5fe-f397-4a89-92ca-79bdca45482e.png">